lightning/src/util/ser_macros.rs

```diff
@@ -287,26 +287,26 @@ macro_rules! read_tlv_fields {
 // `Self { ,,vecfield: vecfield }` which is obviously incorrect. Instead, we have to match here to
 // detect at least one empty field set and skip the potentially-extra comma.
 macro_rules! _init_tlv_based_struct {
-	({}, {$($field: ident),*}, {$($vecfield: ident),*}) => {
-		Ok(Self {
+	($($objs: ident)::*, {}, {$($field: ident),*}, {$($vecfield: ident),*}) => {
```

Comment:
Would `$type` be more accurate than `$objs`? Or cannot it not be used because it is reserved?